sdcm/gemini_thread.py

```diff
@@ -66,50 +66,106 @@ def __init__(self, test_cluster, oracle_cluster, loaders, stress_cmd, timeout=No
+            'async-objects-stabilization-attempts': 10,
+            'max-mutation-retries-backoff': '1s',
+            'max-mutation-retries': 10,
+            'dataset-size': 'large',
+            'oracle-host-selection-policy': 'token-aware',
+            'test-host-selection-policy': 'token-aware',
+            'drop-schema': 'true',
+            'cql-features': 'normal',
+            'fail-fast': 'true',
+            'materialized-views': 'false',
+            'use-server-timestamps': 'true',
+            'use-lwt': 'false',
+            'use-counters': 'false',
+            'max-tables': 1,
+            'max-columns': 16,
+            'min-columns': 8,
+            'max-partition-keys': 6,
+            'min-partition-keys': 2,
+            'max-clustering-keys': 4,
+            'min-clustering-keys': 2,
+
+            'partition-key-distribution': 'normal',  # Distribution for hitting the partition
+            # These two are used to control the memory usage of Gemini
+            'token-range-slices': 512,  # Number of partitions
+            'partition-key-buffer-reuse-size': 100, # Internal Channel Size per parittion value generation
+        }
+
+        self.gemini_oracle_statements_file = f"gemini_oracle_statements_{self.unique_id}.log"
+        self.gemini_test_statements_file = f"gemini_test_statements_{self.unique_id}.log"
+        self.gemini_result_file = f"gemini_result_{self.unique_id}.log"
 
     def _generate_gemini_command(self):
-        seed = self.params.get('gemini_seed')
+        seed = self.params.get('gemini_seed') or random.randint(1, 100)
         table_options = self.params.get('gemini_table_options')
-        if not seed:
-            seed = random.randint(1, 100)
+        log_statements = self.params.get('gemini_log_cql_statements') or False
+
         test_nodes = ",".join(self.test_cluster.get_node_cql_ips())
-        oracle_nodes = ",".join(self.oracle_cluster.get_node_cql_ips()) if self.oracle_cluster else None
-
-        cmd = "./{} --test-cluster={} --outfile {} --seed {} --request-timeout {}s --connect-timeout {}s ".format(
-            self.stress_cmd.strip(),
-            test_nodes,
-            self.gemini_result_file,
-            seed,
-            self.gemini_request_timeout,
-            self.gemini_connect_timeout)
-        if oracle_nodes:
-            cmd += "--oracle-cluster={} ".format(oracle_nodes)
+        oracle_nodes = ",".join(self.oracle_cluster.get_node_cql_ips())
+
+        cmd = f"gemini \
+                --non-interactive \
+                --oracle-cluster=\"{oracle_nodes}\" \
+                --test-cluster=\"{test_nodes}\" \
+                --seed={seed} \
+                --schema-seed={seed} \
+                --profiling-port=6060 \
+                --bind=0.0.0.0:2121 \
+                --outfile=/{self.gemini_result_file} \
+                --replication-strategy=\"{{'class': 'NetworkTopologyStrategy', 'replication_factor': '3'}}\" \
```

Comment:
why we can override this ?

Reply:
Now it shouldn't be possible to override it